src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala

```diff
@@ -608,6 +610,97 @@ class WorkspaceService(protected val userInfo: UserInfo, val dataSource: SlickDa
       }
     }
 
+  def queryEntities(workspaceName: WorkspaceName, entityType: String, requestedQuery: EntityQuery): Future[PerRequestMessage] = {
+    object AttributeStringifier {
+      def apply(attribute: Attribute): String = {
+        attribute match {
+          case AttributeNull => ""
+          case AttributeString(value) => value
+          case AttributeNumber(value) => value.toString()
+          case AttributeBoolean(value) => value.toString()
+          case AttributeEmptyList => ""
+          case AttributeValueList(list) => list.map(apply).mkString(" ")
+          case AttributeEntityReferenceList(list) => list.map(apply).mkString(" ")
+          case AttributeEntityReference(t, name) => name
+        }
+      }
+    }
+
+    object AttributeOrderingAsc extends Ordering[Attribute] {
+      override def compare(x: Attribute, y: Attribute): Int = {
+        (x, y) match {
+          case (AttributeNumber(value1), AttributeNumber(value2)) => value1.compare(value2)
+          case (_, _) => AttributeStringifier(x).compareToIgnoreCase(AttributeStringifier(y))
```

Comment:
I think this matches current functionality with the exception of sorting numbers numerically